webview-ui/src/utils/context-mentions.ts

```diff
@@ -32,21 +32,25 @@ export function insertMention(text: string, position: number, value: string): {
 	return { newValue, mentionIndex }
 }
 
-export function removeMention(text: string, position: number): { newText: string; newPosition: number } {
+export function insertMentionDirectly(text: string, position: number, value: string): { newValue: string; mentionIndex: number } {
 	const beforeCursor = text.slice(0, position)
 	const afterCursor = text.slice(position)
+	const newValue = beforeCursor + "@" + value + " " + afterCursor
+	const mentionIndex = position
+	return { newValue, mentionIndex }
+}
 
-	// Check if we're at the end of a mention
+export function removeMention(text: string, position: number): { newText: string; newPosition: number } {
+	const beforeCursor = text.slice(0, position)
+	const afterCursor = text.slice(position)
 	const matchEnd = beforeCursor.match(new RegExp(mentionRegex.source + "$"))
 
 	if (matchEnd) {
-		// If we're at the end of a mention, remove it
-		const newText = text.slice(0, position - matchEnd[0].length) + afterCursor.replace(" ", "") // removes the first space after the mention
+		const newText = text.slice(0, position - matchEnd[0].length) + afterCursor.replace(" ", "")
 		const newPosition = position - matchEnd[0].length
 		return { newText, newPosition }
 	}
 
-	// If we're not at the end of a mention, just return the original text and position
```

Comment:
can we please revert the comments being removed here? this is useful context i'd like to keep included